strawberryfields/device.py

```diff
@@ -142,18 +142,35 @@ def validate_parameters(self, **parameters: complex) -> None:
         if self.gate_parameters is None:
             return
 
+        def _flatten(iterable: Iterable) -> Generator:
+            for it in iterable:
+                if isinstance(it, Iterable) and not isinstance(it, str):
+                    yield from _flatten(it)
+                else:
+                    yield it
```

Comment:
Why choosing this approach over using an utility function from `itertools`?

Reply:
Oh, I think I answered myself: `itertools` won't flatten deeply nested arrays whereas here you use recursion to do so. Great approach!

Reply:
Thanks, although it's probably a bit overkill to allow for deeply nested arrays, to be honest.